test/stockpile_test.clj

```diff
@@ -129,6 +133,27 @@
          (is (= [] entries))
          (is (not (.exists garbage))))))))
 
+(defn test-discard-entry-to [destination tmpdir q-name]
+  (let [qdir (.resolve tmpdir q-name)
+        newq (stock/create qdir)]
+    (let [entry (store-str newq "foo")
+          [q read-entries] (stock/open qdir conj ())]
+      (is (= [entry] read-entries))
+      (stock/discard q entry destination)
+      (is (= "foo" (String. (Files/readAllBytes destination) "UTF-8"))))))
+
+(deftest discard-to-destination
+  (call-with-temp-dir-path
+   (fn [tmpdir]
+     (test-discard-entry-to (.resolve tmpdir "discarded") tmpdir "q1")
+     (when small-test-fs
```

Comment:
As discussed, I added a more general message when we check for the tinyfs environment variable at the top.